drop-transfer/src/file/id.rs

```diff
@@ -101,6 +101,10 @@ impl FileSubPath {
         self.0.last().expect("Missing last path component")
     }
 
+    pub fn root(&self) -> &String {
+        self.0.first().expect("Missing first path component")
```

Comment:
Sanity check: are we sure just a tokio threat will panic and it will return as an error?

Reply:
It depends. The code logic forbids empty subpaths, hence this is more like an assertion

Reply:
Yep but I am cautious if this could result in a panic. We would handle it via our custom handler though. I can see we have more of these expects and they didn't really manifest so far so I assume it's sort of safe